src/Jackett.Common/Indexers/TorrentHeaven.cs

```diff
@@ -202,12 +198,14 @@ protected override async Task<IEnumerable<ReleaseInfo>> PerformQuery(TorznabQuer
                     var catStr = qCatLink.GetAttribute("href").Split('=')[3].Split('#')[0];
                     var link = new Uri(SiteLink + qDlLink.GetAttribute("href"));
                     var sizeStr = qSize.TextContent;
-                    var dateStr = qDateStr.TextContent.Trim().Replace("Heute", "Today").Replace("Gestern", "Yesterday");
-
+                    var dateStr = qDateStr.TextContent;
+                    var split = dateStr.IndexOf("Uploader", StringComparison.OrdinalIgnoreCase);
```

Comment:
For me is much more clear if you do
```
var dateSplit = dateStr.Split(new[] { "Uploader" }, StringSplitOptions.None);
```
and then you take the parts you need

Reply:
`IndexOf(string)` stops at the first one that's found, and `SubString(start, length)`. Split finds every instance and puts them all in an array.

Reply:
I know your implementation is a bit more performant but I still prefer
`dateStr = dateStr.Split(new[] { "Uploader" }, StringSplitOptions.None).First().Trim()
.Replace("Heute", "Today").Replace("Gestern", "Yesterday");`
When there are no big savings I prefer that it be readable.